sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala

```diff
@@ -314,8 +311,7 @@ case class DataSource(
             catalogTable.get,
             catalogTable.get.stats.map(_.sizeInBytes.toLong).getOrElse(0L))
         } else {
-          new InMemoryFileIndex(
-            sparkSession, globbedPaths, options, partitionSchema)
+          new InMemoryFileIndex(sparkSession, globbedPaths, options, partitionSchema)
```

Comment:
`InMemoryFileIndex` should use the original one because it extends `PartitioningAwareFileIndex` which uses the following line.
```scala
protected val hadoopConf = sparkSession.sessionState.newHadoopConfWithOptions(parameters)
```

Eventually, it created another **case-sensitive** Map from the `CaseInsensitiveMap`. In that case, it will fail to retrieve data from that map because the key value is changed as lowercases already in `CaseInsensitiveMap`.
```scala
def newHadoopConfWithOptions(options: Map[String, String]): Configuration = {
    val hadoopConf = newHadoopConf()
    options.foreach { case (k, v) =>
      if ((v ne null) && k != "path" && k != "paths") {
        hadoopConf.set(k, v)
      }
    }
    hadoopConf
  }
```